logstash-core/spec/logstash/config/source_loader_spec.rb

```diff
@@ -88,11 +94,19 @@ def match?
 
         expect(result.success?).to be_truthy
         expect(result.response.size).to eq(2)
-        expect(result.response).to include(DummySource, AnotherDummySource)
+        expect(result.response.collect(&:pipeline_id)).to include("dummy_source_id", "another_dummy_source_id")
+      end
+
+      context "when duplicate id is returned" do
+        it "fails to return pipeline" do
+          subject.configure_sources([AnotherDummySource.new(settings), AnotherDummySource.new(settings)])
+          result = subject.fetch
+          expect(result.success?).to be_falsey
+        end
       end
     end
 
-    context "when multiple match" do
+    context "when one match" do
```

Comment:
We had 2 tests with the same name, 2 different expectation, I've renamed this one to what is is actually doing.